src/IW/Core/Issue.hs

```diff
@@ -19,4 +19,8 @@ data Issue = Issue
     , issueRepoName :: Text
     , issueLabels   :: SqlArray Text
     } deriving stock (Generic, Show, Eq)
-      deriving anyclass (ToJSON, FromRow, ToRow)
+      deriving anyclass (ToJSON, FromRow)
+
+instance ToRow Issue where
+    toRow Issue{..} = toRow (issueNumber, issueTitle, issueBody, issueUrl, issueOwner, issueRepoName, issueLabels)
```

Comment:
This instance does the work for now. But I find a couple of problems with this instance.

1. It doesn't use the `Id` type. For me, it's just strange when some fields are ignored. In my understanding, `FromRow` and `ToRow` instances should use all fields of the data type and satisfy roundtrip property-test. I had an idea for this particular problem. The solution is to introduce the following data type:
```haskell
data WithId a = WithId
    { withIdId :: Id a
    , withidVal :: a
    }
```
So, `WithId` has only `FromRow` instance, and data types like `Issue` and `Repo` doesn't store `WithId`. But we never introduced such data type in our system because it was awkward to work with.
2. You're using `RecordWildCards` here. In most cases where we use `RecordWildCards` they are perfectly fine. But here there could be a problem: somebody can introduce new fields to the `Issue` data type, and the compiler won't tell you to update this `ToRow` instance. So it could easily become broken. Again, in general case, even this is not a problem, because if you have `FromRow` instance and you have roundtrip property-based tests, you will check this. But since this instance doesn't satisfy this property and in SQL it's okay to have nullable columns so even unit tests won't catch this error, this instance is error-prone.

With `Generic` we automatically get the property of satisfying roundtrip law, and the instances will be updated automatically once you change fields. So, the solution here is to implement custom smart property-based roundtrip tests for such data types. I propose to create an issue about that, and we will discuss and implement it later.

Reply:
Okay got it. I think I understand what we're trying to accomplish here but I'll do some more research and discuss next time we do a conference call